src/main/java/mate/academy/hwspringbootintro/validator/validator/FieldMatchValidator.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.hwspringbootintro.validator.validator;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.Field;
+import mate.academy.hwspringbootintro.exception.RegistrationException;
+import mate.academy.hwspringbootintro.validator.annotation.FieldMatch;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+    private String message;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstFieldName = constraintAnnotation.first();
+        secondFieldName = constraintAnnotation.second();
+        message = constraintAnnotation.message();
+    }
+
+    @Override
+    public boolean isValid(Object obj, ConstraintValidatorContext constraintValidatorContext) {
+        Class<?> clazz = obj.getClass();
```

Comment:
you can simplify the solution
```suggestion
        Object field = new BeanWrapperImpl(value).getPropertyValue(this.field);
        Object fieldMatch = new BeanWrapperImpl(value).getPropertyValue(this.fieldMatch);
        return Objects.equals(field, fieldMatch);
```